### PR TITLE
gracefully release leases

### PR DESCRIPTION
**Pull request by atiratree**

### CODE REVIEW

**Comment by deads2k** [comment]
You can't do this  until this image is being used in payloads. Once it is used in payloads and the old code in openshift-controller-manager is removed, you may do this at wil.

**Comment by atiratree** [comment]
> You can't do this until this image is being used in payloads. Once it is used in payloads and the old code in openshift-controller-manager is removed, you may do this at wil.

Done

File: pkg/cmd/routecontroller/route.go [hunk 1]
```diff
@@ -64,11 +64,11 @@ func RunRouteControllerManager(config *openshiftcontrolplanev1.OpenShiftControll
 			Callbacks: leaderelection.LeaderCallbacks{
 				OnStartedLeading: routeControllerManager,
 				OnStoppedLeading: func() {
-					klog.Fatalf("leaderelection lost")
+					klog.Exit("Route Controller Manager received stop signal: leaderelection lost")
```
[the remark this answers is not in the document]
Comment (atiratree) [reply pkg/cmd/routecontroller/route.go]: yup, 0 exit code should work better as this an intended flow of the program. PR updated

**Pull request merged**